plugins/CoreHome/vue/src/WidgetLoader/WidgetLoader.vue

```diff
@@ -77,8 +77,8 @@ export default defineComponent({
     };
   },
   watch: {
-    widgetParams(parameters: QueryParameters) {
-      if (parameters) {
+    widgetParams(parameters: QueryParameters, oldParameters: QueryParameters) {
+      if (parameters && JSON.stringify(parameters) !== JSON.stringify(oldParameters)) {
```

Comment:
If it's also executed on mount, the mounted() code could just be removed?

Reply:
Wasn't sure if that is always the case or some kind of race condition because of the angularjs <> vue mixup maybe.

Reply:
Hm, I'll investigate as I don't understand exactly what's happening here.

Reply:
I couldn't reproduce this, tried in the dashboard and in a report page. It's possible the parameters object is set to a new object that has the exact same value, maybe when loading the same page? If it's something you could reproduce it would be good to figure out where `JSON.stringify(oldParameters) === JSON.stringify(newParameters)` gets triggered (via `debugger;` for instance). Not doing anything in this case might result in nothing happening when the user tries to reload the current page (ie, user is on Visitors Overview, then clicks Visitors Overview to reload the widgets).

Reply:
@diosmosis It only happens on the admin page and only with disabled developer mode (merged assets).

Reply:
@sgiehl fixed it here: https://github.com/matomo-org/matomo/pull/18672

It happens w/ developer mode enabled too.